components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/APIClient.java

```diff
@@ -147,11 +187,12 @@ private ActionInvocationResponse handleResponse(HttpResponse response) {
             case HttpStatus.SC_SERVICE_UNAVAILABLE:
             case HttpStatus.SC_GATEWAY_TIMEOUT:
                 actionInvocationResponseBuilder.errorLog(
-                        "Failed to execute the action request. Received: " + statusCode);
+                        "Failed to execute the action request for status code " + statusCode + ".");
```

Comment:
This message is wrong. Expects status 200 which is success. So this should denote that the received status code is something else

Reply:
Better say "Failed to execute the action request. Received status code {status}"

Reply:
Updated in